mod/cli/pkg/builder/builder.go

```diff
@@ -156,8 +157,11 @@ func (cb *CLIBuilder[T, ExecutionPayloadT]) defaultRunHandler(
 // InterceptConfigsAndCreateContext except it also sets the server context on
 // the command and the server logger.
 func (cb *CLIBuilder[T, ExecutionPayloadT]) InterceptConfigsPreRunHandler(
-	cmd *cobra.Command, logger log.Logger, customAppConfigTemplate string,
-	customAppConfig interface{}, cmtConfig *cmtcfg.Config,
+	cmd *cobra.Command,
+	logger log.AdvancedLogger[any, sdklog.Logger],
```

Comment:
see if we can do LoggerT for the advanced logger otherwise make the sdk logger a generic type